### PR TITLE
Add recipe for Python API of DFT-D3

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by awvwgk** [comment]
@conda-forge/help-python-c This recipe is ready for review.

File: recipes/dftd3-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - pip
+    - simple-dftd3
+    - libblas
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux and not aarch64]
+  run:
+    - python
+    - cffi
+    - numpy
+    - {{ pin_compatible('simple-dftd3', max_pin='x.x') }}
```
Comment (xhochy) [inline recipes/dftd3-python/meta.yaml:34]: You should instead rather add a `run_exports` to `simple-dftd3`

File: recipes/dftd3-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  sha256: 79b507a0d892f19222e1403f4c7ebb98ce9eb96b3b8d4821ba5631aa9ac756be
+
+build:
+  number: 0
+  skip: True  # [win or (osx and (python_impl == 'cpython' and py<38))]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - meson 0.55.*
```
Comment (xhochy) [inline recipes/dftd3-python/meta.yaml:19]: What's the reason for this hard meson pin? Maybe add a comment here.

File: recipes/dftd3-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - python
+    - cffi
+    - pkgconfig
+  host:
+    - python
+    - pip
+    - simple-dftd3
+    - libblas
+    - llvm-openmp  # [osx]
+    - libgomp  # [linux and not aarch64]
```
Comment (xhochy) [inline recipes/dftd3-python/meta.yaml:29]: What is the cause of not needing `libgomp` on `aarch64`?

File: recipes/dftd3-python/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  number: 0
+  skip: True  # [win or (osx and (python_impl == 'cpython' and py<38))]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - meson 0.55.*
+    - python
+    - cffi
+    - pkgconfig
```
Comment (xhochy) [inline recipes/dftd3-python/meta.yaml:22]: You probably want the `pkg-config` executable and not the Python package?
```suggestion
    - pkg-config
```

File: recipes/dftd3-python/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    - numpy
+    - {{ pin_compatible('simple-dftd3', max_pin='x.x') }}
+
+
+test:
+  imports:
+    - dftd3
+  requires:
+    - pytest
+    - setuptools
```
Comment (xhochy) [inline recipes/dftd3-python/meta.yaml:42]: A bit curious why you need `setuptools` here. That isn't a typical test dependency.

Comment (awvwgk) [reply recipes/dftd3-python/meta.yaml:40]: I'm using `pkg_resources.parse_version` in [one test](https://github.com/awvwgk/simple-dftd3/blob/b33ae37eed3f70c0cef5bad58b31904e6c6c3941/python/dftd3/test_libdftd3.py#L18). I could probably just split the string and save me the trouble in a future version.

File: recipes/dftd3-python/meta.yaml (hunk 3 above)
Comment (awvwgk) [reply recipes/dftd3-python/meta.yaml]: Found it in the maintainer knowledge base originally this way, I see it was removed from there a while ago: https://github.com/conda-forge/conda-forge.github.io/pull/976/commits/aea88f56f0d05705c731407c0bc983efbc95474a. I just tried running with OpenMP dependency and this seems to work fine for the CFFI module.

File: recipes/dftd3-python/meta.yaml (hunk 2 above)
Comment (awvwgk) [reply recipes/dftd3-python/meta.yaml]: 0.56.* and 0.57.* were quite broken lately for most of my projects, seems to be fixed with the 0.58.* release now.

File: recipes/dftd3-python/meta.yaml (hunk 1 above)
Comment (awvwgk) [reply recipes/dftd3-python/meta.yaml]: Updated in upstream feedstock

**Pull request merged**